src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,209 @@
+                PreparedStatement getAllCarsStatement =
+                        connection.prepareStatement(getCarsAndManufacturersQuery)) {
+            ResultSet resultSet = getAllCarsStatement.executeQuery();
+            while (resultSet.next()) {
+                cars.add(parseCarWithManufacturer(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get All cars", e);
+        }
+        for (Car car : cars) {
+            car.setDriverList(getDriversForCar(car.getId()));
+        }
+        return cars;
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long id) {
+        String getAllByDriverQuery = "SELECT c.id, c.model, c.manufacturer_id, m.name, m.country "
+                + "FROM cars_drivers cd "
+                + "JOIN cars c ON cd.car_id = c.id "
+                + "JOIN manufacturers m ON m.id = c.manufacturer_id "
+                + "WHERE cd.driver_id = ? and c.is_deleted = FALSE;";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getAllCarsByDriverStatement =
+                        connection.prepareStatement(getAllByDriverQuery)) {
+            getAllCarsByDriverStatement.setLong(1, id);
+            ResultSet resultSet = getAllCarsByDriverStatement.executeQuery();
+            while (resultSet.next()) {
+                cars.add(parseCarWithManufacturer(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get all cars for one driver by id" + id, e);
+        }
+        return cars;
+    }
+
+    private void deleteOldRelationsCarsToDrivers(Car car) {
+        String deleteOldRelations = "DELETE FROM cars_drivers WHERE car_id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement deleteRelationsStatement =
+                        connection.prepareStatement(deleteOldRelations)) {
+            deleteRelationsStatement.setLong(1, car.getId());
+            deleteRelationsStatement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete car relations to drivers: " + car, e);
+        }
+    }
+
+    private Car parseCarWithManufacturer(ResultSet resultSet) throws SQLException {
+        Car car = new Car();
+        car.setModel(resultSet.getString("model"));
+        car.setId(resultSet.getObject("id", Long.class));
+        Manufacturer manufacturer = new Manufacturer();
+        manufacturer.setId(resultSet.getObject("manufacturer_id", Long.class));
+        manufacturer.setName(resultSet.getString("name"));
+        manufacturer.setCountry(resultSet.getString("country"));
+        car.setManufacturer(manufacturer);
+        return car;
+    }
+
+    private List<Driver> getDriversForCar(Long id) {
+        String getDriverListQuery = "SELECT d.id, d.name, d.license "
```

Comment:
In init_db.sql column name in licence_number
```suggestion
        String getDriverListQuery = "SELECT d.id, d.name, d.license_number "
```